MinGW compile error: 'S_IFSOCK' undeclared

Building vsgXchange with MinGW fails with the following error:
```
home/abuild/rpmbuild/BUILD/vsgXchange-0.0.1+20211222+git.145b2d9/src/ktx/libktx/filestream.c: In function 'ktxFileStream_getsize':
[  116s] /home/abuild/rpmbuild/BUILD/vsgXchange-0.0.1+20211222+git.145b2d9/src/ktx/libktx/filestream.c:327:38: error: 'S_IFSOCK' undeclared (first use in this function)
[  116s]   327 |     if (ftype == S_IFIFO || ftype == S_IFSOCK)
```

--- a/src/ktx/libktx/filestream.c
+++ b/src/ktx/libktx/filestream.c
@@ -50,6 +50,8 @@
   #define S_IFIFO _S_IFIFO
   #define S_IFSOCK 0xC000
   typedef unsigned short mode_t;
+#elif defined(__MINGW32__)
+  #define S_IFSOCK 0xC000
 #endif
 
 #define KTX_FILE_STREAM_MAX (1 << (sizeof(ktx_off_t) - 1) - 1)

